src/core/scalebar/qgsnumericscalebarrenderer.h

```diff
@@ -48,7 +48,7 @@ class CORE_EXPORT QgsNumericScaleBarRenderer: public QgsScaleBarRenderer
   private:
 
     //! Returns the text for the scale bar or an empty string in case of error
-    QString scaleText( double scale ) const;
+    QString scaleText( const QgsScaleBarSettings &settings, double scale ) const;
```

Comment:
I'd switch the parameter ordering around, having double scale first then const QgsScaleBarSettings &settings. No strong feelings though, feel free to ignore.